### PR TITLE
Move to PBJ Account

### LINKED ISSUE #6104: Use the generated `Account` PBJ Object instead of Account and AccountBuilder implementations

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #6104 

- Replace `Account` and `AccountBuilder` and its implementations with the PBJ Account Object
- Move `HederaKey` to `PreHandleResult`
- Change the keys in `PreHandleContext` to use PBJ key
- Converts all keys in `PreHandleContext` to `HederaKey` in `PreHandleResult`
- Change to use PBJ Keys in all types of `AccountStore`, `TokenStore`, `TopicStore`, `ScheduleStore`

### CODE REVIEW

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java:76]: I believe we're currently relying on `Util.asHederaKey()` to [validate this key](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java#L38).

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:145]: Does this need the`key == null || key.key().kind().equals(KeyOneOfType.UNSET)` pattern you use [here](https://github.com/hashgraph/hedera-services/blob/24a13c339ae04929be0ba659b59bbf6f445506ec/hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java#L186)?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:91]: Validation concern as above.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java:78]: I believe this would filter `KeyList{}` from the input `Set<Key>` (again because `Utils.asHederaKey()` does validations). I.e. this _could_ e.g. change signing requirements for a transaction involving `0.0.800`. 🤔 

Probably doesn't matter given all this is temporary code, but worth noting.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java:76]: yeah right. I was adding `isEmpty` method in `PreHandleContext` and use it in all places. Will probably add `isValid` as well ?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:91]: Added validation to `requireKey`

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:145]: Thanks @tinker-michaelj  ! Added `!isEmpty` check

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java]: Thanks @tinker-michaelj ! Added `isValid` method to `requireKey` method

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java:78]: I think it will not filter, it will add an empty `KeyList` when converting to PBJ right ?

Comment (tinker-michaelj) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java:78]: Ah true, there should be an earlier `PreHandleException` in the case of an immutable account. Good point.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java
Comment (tinker-michaelj) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java:69]: Isn't this unreachable, b/c if it was `true`, then we would have returned `false` from the `isEmpty()` condition?

Comment (mhess-swl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java:69]: Swap to avoid potential NPE?
```suggestion
        if (KeyOneOfType.UNSET.equals(key.kind()) {
```

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java:146]: Extra comment line
```suggestion
```

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/ReadableAccountStoreTest.java
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/ReadableAccountStoreTest.java:154]:
```suggestion
        subject = new ReadableAccountStore(readableStates);
```

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java:76]: Is there a reason to mock `transferAccount` and `deleteAccount` instead of using real account objects?

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java:111]: Assigning a single base variable inside of multiple tests like this makes it hard to run tests in parallel. Not a huge issue, but we may want to revisit this

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java
Comment (tinker-michaelj) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java:64]: Seems like making this `@Nullable final Key pbjKey` and return `false` on `null` would save some typing in other places.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java
Comment (tinker-michaelj) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:187]: As noted above, I'd suggest make `isValid(null)` just return `false` and simplify this condition.

**Review by tinker-michaelj — APPROVED** [review]
I have some confusion and/or doubts about `PreHandleContext.require*()` method variants, but they weren't introduced in this PR, and will be much easier to solve in a smaller change-set in any case.

LGTM, tyvm for grinding through this @Neeharika-Sompalli!

File: hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/state/MapReadableKVState.java
Comment (jsync-swirlds) [inline hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/state/MapReadableKVState.java:40]: Shouldn't need the `? super` anymore, it was there for Comparable, which is no longer required.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java
Comment (rbair23) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java:26]: I'm not sure this is quite right, or in quite the right place. I will be looking through the review for which modules need to use this class. Maybe it doesn't have to be public in the SPI? Unless there is an obvious answer I don't propose holding up the PR, but I just wanted to flag that it seems a little off.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/ReadableTopicStoreTest.java
Comment (rbair23) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/ReadableTopicStoreTest.java:81]: I don't know why we have tests for toString (or why we implement the method either). It shouldn't ever be called in production (performance impact), and in development we can use a debugger. It isn't clear to me why we have this code.

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ReadableScheduleStore.java
Comment (rbair23) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ReadableScheduleStore.java:68]: What does "sanityRestoredToPbj" do?

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java
Comment (rbair23) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java:78]: What does `isEmpty` do if you pass it a null key? Does it check for null too? I'm wondering if the null check here is being done twice.

File: hedera-node/hedera-schedule-service-impl/src/main/java/module-info.java
Comment (rbair23) [inline hedera-node/hedera-schedule-service-impl/src/main/java/module-info.java:13]: Does this mean test fixtures is required by production code?

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCreateHandlerTest.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCreateHandlerTest.java:40]: Thanks @mhess-swl ! Nothing, typo while refactoring all tests 🙂

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java:69]: I don't think its possible since PBJ generates it 🙂 . If nothing is set `UNSET` should be present. But I don't mind making this change

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java:111]: Thanks for the suggestion @mhess-swl . I will keep it in mind and add different methods to separate them out in next PRs.

File: hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/state/MapReadableKVState.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/state/MapReadableKVState.java:40]: I thought Artem's PR did not make it into the platform sdk `0.37.0-adhoc.xc76224af`. Will remove when next platform sdk is released

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/ReadableAccountStoreTest.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/ReadableAccountStoreTest.java:154]: Yes done. Thanks

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java:78]: Moved null check into `isEmpty`. If the key is null the `isEmpty` returns true.

File: hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/state/MapReadableKVState.java
Comment (jsync-swirlds) [reply hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/state/MapReadableKVState.java:40]: Ahh, missed that this is still a little behind platform.
Yes, that would be a separate PR, then.

**Review by mhess-swl — DISMISSED** [review]

**Comment by mhess-swl** [comment]
Tons of work 🥴 thanks Neeha!

**Review by jsync-swirlds — DISMISSED** [review]

**Review by iwsimon — DISMISSED** [review]
LGTM

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java
Comment (jsync-swirlds) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java:46]: Nit: pbjKey is `@Nullable`, as it does permit null (line 47 checks for it and returns true)

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java]: Thanks Joseph. Fixed

**Review by jsync-swirlds — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by jeromy-cannon — APPROVED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

**Review by jsync-swirlds — APPROVED** [review]

**Pull request merged**
